### PR TITLE
Redesign send request page

### PR DESCRIPTION
**Pull request by TheBestTvarynka**

### CODE REVIEW

File: backend/core/src/main/java/com/feed_grabber/core/report/ReportService.java [hunk 1]
```diff
@@ -54,4 +61,32 @@ public String generateReport(UUID requestId) throws NotFoundException, JsonProce
...
+
+    public void hasAccess(UUID requestId, UUID userId, String role) {
+        if (role.equals(ROLE_COMPANY_OWNER) || role.equals(ROLE_HR)) {
+            return;
+        }
+        Optional<Request> request = requestRepository.findByIdAndTargetUser(requestId, userId);
+        if (request.isEmpty() || !request.get().getSendToTarget()) {
+            throw new AccessDeniedException("You have not enough permissions to view this report");
+        }
+    }
```
Comment (TheBestTvarynka) [inline backend/core/src/main/java/com/feed_grabber/core/report/ReportService.java:72]: This method checks if the user has access to the report.
We grant access only in three cases:
1-2) if the user is a company owner or HR
3) if the user is a target user of request and field `sendToRequest` is true

File: backend/core/src/main/java/com/feed_grabber/core/request/RequestRepository.java [hunk 2]
```diff
@@ -24,5 +26,13 @@
 
     List<Request> findAllByQuestionnaireId(UUID id);
 
+    @Query("select r from Request r where r.closeDate is not NULL and r.questionnaire.company.id = :companyId")
+    List<Request> findAllReports(UUID companyId);
```
Comment (TheBestTvarynka) [inline backend/core/src/main/java/com/feed_grabber/core/request/RequestRepository.java:30]: Search all available reports for the company owner/HR.

File: backend/core/src/main/java/com/feed_grabber/core/request/RequestRepository.java [hunk 3]
```diff
@@ -24,5 +26,13 @@
...
+
+    @Query("select r from Request r " +
+            "where r.closeDate is not null " +
+            "and r.sendToTarget = true and r.targetUser.id = :userId")
+    List<Request> findReportsForEmployee(UUID userId);
```
Comment (TheBestTvarynka) [inline backend/core/src/main/java/com/feed_grabber/core/request/RequestRepository.java:35]: Search available reports for simple user (employee).

File: backend/core/src/main/java/com/feed_grabber/core/response/ResponseController.java [hunk 4]
```diff
@@ -41,8 +46,11 @@
     }
 
     @GetMapping("/users")
-    @Secured(value = {ROLE_COMPANY_OWNER, ROLE_HR})
+    // @Secured(value = {ROLE_COMPANY_OWNER, ROLE_HR})
     public AppResponse<List<UserResponseShortDto>> getRespondentsShortInfo(@RequestParam UUID requestId) {
+        final String role = TokenService.getRoleName();
+        final UUID userId = TokenService.getUserId();
+        reportService.hasAccess(requestId, userId, role);
```
Comment (TheBestTvarynka) [inline backend/core/src/main/java/com/feed_grabber/core/response/ResponseController.java:57]: Also, I change security here (corresponds to the logic described above).

File: frontend/src/sagas/report/sagas.ts [hunk 5]
```diff
@@ -70,11 +71,23 @@ function* loadUsersReports(action) {
     }
 }
 
+function* loadAwailableReports(action) {
+    try {
+        const res = yield call(apiClient.get, '/api/report/all');
+        yield put(loadReportsRoutine.success(res.data.data));
+    } catch (err) {
+        console.log(err);
```
Comment (marchuk0) [inline frontend/src/sagas/report/sagas.ts]: delete console.log() please

File: backend/core/src/main/java/com/feed_grabber/core/request/RequestRepository.java (hunk 2 above)
Comment (marchuk0) [reply backend/core/src/main/java/com/feed_grabber/core/request/RequestRepository.java:30]: But why method called "findAllReports" returns list of request?

File: backend/core/src/main/java/com/feed_grabber/core/request/RequestRepository.java (hunk 3 above)
Comment (marchuk0) [reply backend/core/src/main/java/com/feed_grabber/core/request/RequestRepository.java:35]: same with this method

**Review by marchuk0 — CHANGES_REQUESTED** [review]

**Review by marchuk0 — APPROVED** [review]

**Pull request merged**